### PR TITLE
@craigspaeth (cc @kanaabe): Large user refactor

### PR DESCRIPTION
**Pull request by craigspaeth**
Closes https://github.com/artsy/positron/issues/156

The Positron API now does less stuff with users and holds a smaller + flatter user document while the Positron client is responsible for all authing and holding access tokens (essentially making the Positron users collection just a cache, but still allows us to decorate user docs in the future).

Along-side this shift of responsibilities we're dropping v2 orchestration-ey things in favor of the client searching API v1 for users. This combined with the fact that "impersonate" will sync a user to Positron on the fly allows us to drop the silly contributors panel b/c for all intents and purposes Artsy users are Positron users.

In short I'm hoping these changes will make the whole Positron user situation simpler and safer.

### CODE REVIEW

**Comment by kanaabe** [comment]
Awesome!
